website/blog/2024-06-25-use-a-framework-to-build-react-native-apps.md

```diff
@@ -0,0 +1,68 @@
+---
+title: 'Use a framework to build React Native apps'
+authors: [cortinico]
+tags: [announcement]
+date: 2024-06-25
+---
+
+At [React Conf](https://www.youtube.com/live/0ckOUBiuxVY?si=pU4qP4eB5iWfY0IG&t=2320), we updated our guidance on the best tool to get started building React Native apps: a **React Native framework** - a toolbox with all the necessary APIs to let you build production-ready apps.
+
+Using React Native frameworks, such as Expo, is now the **recommended** approach to create new apps.
+
+In this blogpost we want to walk you through what they are in detail and what they mean for you as a React Native developer starting a new project.
+
+<!-- truncate  -->
+
+## What is a React Native framework?
+
+If you’ve been building production-ready apps, you probably know that there is a set of common problems you will need to solve sooner or later.
+
+When building any application on either web or native, you probably want your users to navigate through different screens, fetch data, and store the state of your user. But for native apps there is even more to deal with: you need tools to upgrade your native code between React Native versions, manage compatible versions of all of your dependencies, and deal with native build tools.
+
+It takes a village to bring an app from idea to production without the right tools.
+
+We want you to focus on writing beautiful applications and features for your users, and not solving those common problems over and over.
+
+That’s why we believe that the best way for you to experience React Native is through a framework that offers a toolbox with all the necessary tools you need to build production-ready applications.
+
+We’ve found that **you’re either using a framework or you’re building your own framework**.
+
+There is nothing wrong with building your own framework, by crafting your own solutions for routing, navigation, deploying, and so on. Major corporations like Meta and Microsoft build their own frameworks internally to integrate deeply into their brownfield apps. But we believe that most people will be better off by using an existing framework.
```

Comment:
Not sure it qualifies as a framework but maybe Shopify also builds one?https://shopify.engineering/shopify-tophat-mobile-developer-testing

Reply:
> Not sure it qualifies as a framework but maybe Shopify also builds one?[shopify.engineering/shopify-tophat-mobile-developer-testing](https://shopify.engineering/shopify-tophat-mobile-developer-testing)

I won't call this a framework but more of a testing methodology (at least from what I can read from the blogpost).